### PR TITLE
DBaaS Routing + NavTabs component

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

This cleans up the DBaaS routing a bit, and also adds a new `<NavTabs />` component which aims to addressing the following (subtle) issues with tabbed navigation:

- Changing tabs pushes _two_ events to history, so you have to click "Back" twice to get to the previous tab.
- Bogus routes are accepted (the first tab is selected by default, but the bogus path remains in the URL).
- If any part of the path is bogus, the first tab is always selected, even if part of the part is correct (e.g. `/profile/keys/bogus` displays the first tab, "Display", rather than the "SSH Keys" tab, which would be expected).

Most importantly, there's a lot of duplicated code which has been following us around for a while. (How many times have each of us copy/pasted the comment: `/* NB: These must correspond to the routes inside the Switch */`?)

This is an attempt to simplify things behind a single component. 

## Notes to Reviewers
I used the new component in DBaaS as well as Profile so we could get a feel for how further refactoring will go. So please check these two features extensively, throwing whatever you can at the URL routing and make sure it does what you'd expect.

### CODE REVIEW

File: packages/manager/src/components/TabLinkList/TabLinkList.tsx [hunk 1]
```diff
@@ -72,16 +73,23 @@ export const TabLinkList: React.FC<CombinedProps> = props => {
...
-          to={tab.routeName}
-        >
-          {tab.title}
-        </Tab>
-      ))}
+      {tabs.map((tab, _index) => {
+        // @todo: remove this if we use NavTab widely.
+        const extraTemporaryProps: any = props.noLink
+          ? {}
+          : { as: Link, to: tab.routeName };
```
Comment (johnwcallahan) [inline packages/manager/src/components/TabLinkList/TabLinkList.tsx:80]: This is what was causing the double history event (this + the `navToURL` function). We can remove this if we use NavTabs everywhere.

**Review by tiffwong — APPROVED** [review]
I tried to break it without any luck 👍

**Review by WilkinsKa1 — COMMENTED** [review]
This is working really well/as advertised! The one issue I noticed, which may be a separate issue/isolated to me (but I am able to reproduce consistently), is if I try to go to `/profile/lish/dfgh`, I get an infinite loading state that never redirects. This doesn't occur for other tabs in this section.

File: packages/manager/src/components/NavTabs/NavTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import SuspenseLoader from 'src/components/SuspenseLoader';
+import TabLinkList from '../TabLinkList/TabLinkList';
+import TabPanel from 'src/components/core/ReachTabPanel';
+
+export interface NavTab {
+  title: string;
+  routeName: string;
+  component:
+    | React.ComponentType
+    | React.LazyExoticComponent<React.ComponentType>;
```
Comment (Jskobos) [inline packages/manager/src/components/NavTabs/NavTabs.tsx:15]: dear god what has TypeScript been smoking

**Review by Jskobos — APPROVED** [review]
I like this a lot

**Comment by Jskobos** [comment]
> This is working really well/as advertised! The one issue I noticed, which may be a separate issue/isolated to me (but I am able to reproduce consistently), is if I try to go to `/profile/lish/dfgh`, I get an infinite loading state that never redirects. This doesn't occur for other tabs in this section.

This is in prod too, it's due to loose matching on line 56 of AuthenticationWrapper (I think).

**Comment by Jskobos** [comment]
Going to put up a PR for the Lish thing

**Pull request merged**